04-onion-routing.md

```diff
@@ -238,11 +236,47 @@ Field descriptions:
      leaking its position in the route.
 
    * `padding`: This field is for future use and also for ensuring that future non-0-`realm`
-     `per_hop`s won't change the overall `hops_data` size.
+     `hop_data`s won't change the overall `hop_payloads` size.
+
+When forwarding HTLCs, nodes MUST construct the outgoing HTLC as specified
+within `hop_data` above; otherwise, deviation from the specified HTLC
+parameters may lead to extraneous routing failure.
+
+### `tlv_payload` payload format
+
+This is a more flexible format, which avoids the redundant `short_channel_id` field for the final node. 
+
+1. tlv: `tlv_payload`
+2. types:
+    1. type: 0 (`destination_signal`)
+    2. data:
+	    * Empty
+    1. type: 2 (`amt_to_forward`)
+    2. data:
+        * [`integer`:`amt_to_forward`]
```

Comment:
Still not clear what `integer` is from strictly reading this document. I think using `varint` here makes it more clear when combined with #622.